--- autumn-harvest/src/scheduler.rs
+++ autumn-harvest/src/scheduler.rs
@@ -509,6 +509,7 @@

     for level in dag.definition.execution_levels() {
         let tasks = level.iter().map(|task_index| {
+            /// Avoid an unnecessary heap allocation of `DagTask` per task when `&DagTask` works.
             let task = &dag.definition.tasks()[*task_index];
             let upstream_statuses = task
                 .upstreams
